[configure] Rewrite and modularization of configure [1/n] (#14189)
coq/coq: dev/doc/INSTALL.make.md

    coqtop -coqlib <new directory>

# FLambda Options

> You can tweak optimizations flags sent to the OCaml optimizing

Reviewer: ```suggestion
You can tweak optimization flags passed to the OCaml optimizing
```

> You can tweak optimization flags passed to the OCaml optimizing

Reviewer 2: ```suggestion
You can tweak the optimization flags passed to the OCaml optimizing
```

compiler. Coq's default is:

    -flambda-opts `-O3 -unbox-closures`

> set in Coq's toplevel `dune` file. Feel free to try with a different

Reviewer 2: ```suggestion
Which is set in Coq's toplevel `dune` file. Feel free to try with a different
```

Reviewer 2: ```suggestion
set in Coq's toplevel `dune` file. Feel free to try a different
```

combination of flags. You can read more at
https://caml.inria.fr/pub/docs/manual-ocaml/flambda.html

There is a known problem with certain OCaml versions and

> `native_compute`, that will make compilation to require

Reviewer 2: ```suggestion
`native_compute`, that will make compilation require
```

> a large amount of RAM (>= 10GiB) in some particular files.

Reviewer 2: ```suggestion
a large amount of RAM (>= 10GiB) for some particular files.
```

> We recommend disabling native compilation (`-native-compiler no`)
> with flambda unless you use OCaml >= 4.07.0.

Reviewer: ```suggestion
We recommend disabling native compilation (`-native-compiler no`)
with flambda if you use OCaml < 4.07.0.
```
